### PR TITLE
Fix elasticsearch example volume descriptors

### LINKED ISSUE #26257: Elasticsearch example fails on Kubernetes 1.1.x
The elasticsearch example (production cluster) fails on Kubernetes 1.1.x due to an invalid volume decription in the replication controller files.

```bash
$ kubectl create -f examples/elasticsearch/production_cluster/es-master-rc.yaml
error validating "examples/elasticsearch/production_cluster/es-master-rc.yaml": error validating data: found invalid field source for v1.Volume; if you choose to ignore these errors, turn validation off with --validate=false
```

```bash
$ kubectl create -f examples/elasticsearch/production_cluster/es-client-rc.yaml
error validating "examples/elasticsearch/production_cluster/es-client-rc.yaml": error validating data: found invalid field source for v1.Volume; if you choose to ignore these errors, turn validation off with --validate=false
```

```bash
$ kubectl create -f examples/elasticsearch/production_cluster/es-data-rc.yaml
error validating "examples/elasticsearch/production_cluster/es-data-rc.yaml": error validating data: found invalid field source for v1.Volume; if you choose to ignore these errors, turn validation off with --validate=false
```

I was able to fix this by updating the volume description (PR incoming).
```diff
-      - name: storage
-        source:
-          emptyDir: {}
+      - emptyDir:
+          medium: ""
+        name: "storage"
```

### PR DESCRIPTION
**Pull request by koep**
fixes #26257
[Analytics]()

### CODE REVIEW

File: examples/elasticsearch/production_cluster/es-client-rc.yaml [hunk 1]
```diff
@@ -47,6 +47,6 @@ spec:
         - mountPath: /data
           name: storage
       volumes:
-      - name: storage
-        source:
-          emptyDir: {}
+      - emptyDir:
+          medium: ""
```
Comment (bgrant0607) [inline examples/elasticsearch/production_cluster/es-client-rc.yaml]: Thanks. `medium` is unnecessary, and it reads better if `name` is first.
```
      - name: storage
        emptyDir: {}
```

**Comment by koep** [comment]
@bgrant0607 fixed!

**Comment by koep** [comment]
@jayunit100 

I think the do-not-merge tag was added because I made a mistake while updating my branch a few days ago.

**Comment by bgrant0607** [comment]
ok to test

**Comment by bgrant0607** [comment]
LGTM

Sorry for the delay

**Pull request merged**